app/Http/Controllers/TaskController.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Enums\TaskStatusEnum;
+use App\Http\Requests\TaskUpdateRequest;
+use App\Http\Resources\TaskDetailsResource;
+use App\Models\Task;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Redirect;
+use Inertia\Inertia;
+use Inertia\Response;
+
+class TaskController extends Controller
+{
+
+    /**
+     * Display the list of tasks
+     */
+    public function show(Request $request): Response
+    {
+        return Inertia::render('Dashboard/Index', [
+            'tasks' => TaskDetailsResource::collection(Task::with("user")->get())->toArray($request),
+        ]);
+    }
+
+    /**
+     * create the task.
+     */
+    public function create(TaskUpdateRequest $request): RedirectResponse
+    {
+        Task::create($request->validated());
+        return Redirect::route('dashboard');
+    }
+
+    /**
+     * Update the task status to `in progress` and attribute it to the user.
+     */
+    public function takeOn(Request $request, Task $task): RedirectResponse
+    {
+        $task->user_id = $request->user()->id;
```

Comment:
we can add a middleware for that on the request or a check here. Something like this could cover the basic case:

```
class ValidateTaskBelongsToUserMiddleware
{
    public function handle(Request $request, Closure $next): Response
    {
        $task = Task::query()->findOrFail($request->route('task'));

        if (!empty($task->user_id)) {
            abort(404); // could also return an error info
        }

        return $next($request);
    }
}
```

On the frontend we also need to hide that button ( the validation there is different right now ), and display an error if the user somehow manages to send the request.

-> I like the idea of this type of notification as well: https://github.com/euvl/vue-notification

We could take a step further and add more validations on that tho. for example, if a user will get paid for a task maybe the owner of shouldn't be able to edit all the info as well no?

We could add a requester on the task - Mainly the user who created the task and he could change specific info like deadline, type of task ( need to add the field as well ) and maybe even the description of it